### PR TITLE
Don't process std out/err of canceled pips

### PR DESCRIPTION
**Pull request by almiliMSFT**

### CODE REVIEW

File: Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs [hunk 1]
```diff
@@ -1521,7 +1521,7 @@ private void LogCommandLineTooLong(SandboxedProcessInfo info)
                 }
             }
 
-            bool shouldPersistStandardError = errorOrWarnings || m_pip.StandardError.IsValid;
+            bool shouldPersistStandardError = !canceled && (errorOrWarnings || m_pip.StandardError.IsValid);
```
Comment (mpysson) [inline Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs]: >!canceled [](start = 46, length = 9)

Do we differentiate resource exhaustion cancelled vs. timeout cancelled? I think we'd only want to mask standard output stream logging for the first.

Comment (almiliMSFT) [reply Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs]: From line 1340:
```csharp
bool canceled = result.Killed && cancellationToken.IsCancellationRequested;
```
When a pip times out, `cancellationToken.IsCancellationRequested` is false; it is true only when the pip is explicitly canceled, either due to resource exhaustion or Ctrl+C (and in neither case I don't think we need to save std out/err)

Comment (Kr1sso) [reply Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs]: Agreed, we also make sure to set result.killed to `true` in both of those cases.

Comment (mpysson) [reply Public/Src/Engine/Processes/SandboxedProcessPipExecutor.cs]: Cool. As long as we don't skip processing the output streams on timeout I think the change is good. Thanks for checking.

**Pull request merged**